tests/Generator/Resolver/Value/Chainable/FixtureMethodCallReferenceResolverTest.php

```diff
@@ -0,0 +1,215 @@
+use Prophecy\Argument;
+
+/**
+ * @covers \Nelmio\Alice\Generator\Resolver\Value\Chainable\FixtureMethodCallReferenceResolver
+ */
+class FixtureMethodCallReferenceResolverTest extends \PHPUnit_Framework_TestCase
+{
+    public function testIsAChainableResolver()
+    {
+        $this->assertTrue(is_a(FixtureMethodCallReferenceResolver::class, ChainableValueResolverInterface::class, true));
+    }
+
+    /**
+     * @expectedException \Nelmio\Alice\Throwable\Exception\UnclonableException
+     */
+    public function testIsNotClonable()
+    {
+        clone new FixtureMethodCallReferenceResolver();
+    }
+
+    public function testWithersReturnNewModifiedInstance()
+    {
+        $resolver = new FixtureMethodCallReferenceResolver();
+        $newResolver = $resolver->withValueResolver(new FakeValueResolver());
+
+        $this->assertEquals(new FixtureMethodCallReferenceResolver(), $resolver);
+        $this->assertEquals(new FixtureMethodCallReferenceResolver(new FakeValueResolver()), $newResolver);
+    }
+
+    public function testCanResolveMethodCallReferenceValues()
+    {
+        $resolver = new FixtureMethodCallReferenceResolver();
+
+        $this->assertTrue($resolver->canResolve(new FixtureMethodCallValue(new FakeValue(), new FunctionCallValue('method'))));
+        $this->assertFalse($resolver->canResolve(new FakeValue()));
+    }
+
+    /**
+     * @expectedException \Nelmio\Alice\Throwable\Exception\Generator\Resolver\ResolverNotFoundException
+     * @expectedExceptionMessage Expected method "Nelmio\Alice\Generator\Resolver\Value\Chainable\FixtureMethodCallReferenceResolver::resolve" to be called only if it has a resolver.
+     */
+    public function testCannotResolveValueIfHasNoResolver()
+    {
+        $value = new FixtureMethodCallValue(new FakeValue(), new FunctionCallValue('method'));
+        $resolver = new FixtureMethodCallReferenceResolver();
+        $resolver->resolve($value, new FakeFixture(), ResolvedFixtureSetFactory::create(), [], new GenerationContext());
+    }
+
+    public function testReturnsSetWithResolvedValue()
+    {
+        $value = new FixtureMethodCallValue(
+            $reference = new FakeValue(),
+            $functionCall = new FunctionCallValue('getFoo', [
+                $arg1 = new FakeValue(),
+            ])
+        );
+        $fixture = new FakeFixture();
+        $set = ResolvedFixtureSetFactory::create(new ParameterBag(['foo' => 'bar']));
+        $scope = ['val' => 'scopie'];
+        $context = new GenerationContext();
+        $context->markIsResolvingFixture('foo');
+
+        $valueResolverContext = new GenerationContext();
+        $valueResolverContext->markIsResolvingFixture('foo');
+        $valueResolverContext->markAsNeedsCompleteGeneration();
+
+        $dummyProphecy = $this->prophesize(DummyWithGetter::class);
+        $dummyProphecy->getFoo('resolved_argument')
+            ->shouldBeCalled()
+            ->willReturn('resolved_value');
```

Comment:
the semi-column should be on the new line (yeah I know not everyone likes it, but let's be consistent :P)